src/main/java/com/google/starfish/servlets/UploadGNoteServlet.java

```diff
@@ -0,0 +1,122 @@
+package com.google.starfish.servlets;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;  
+import javax.servlet.http.Cookie;  
+import com.google.starfish.services.LabelService;
+import com.google.starfish.services.MiscNoteLabelService;
+import javax.sql.DataSource;
+import java.sql.Connection;  
+import java.sql.SQLException;  
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Date;
+import java.util.Calendar;
+
+/**
+ * Servlet to handle note uploading and adding associated labels to the database
+ * for gnotes from the chrome extension
+ */
+@WebServlet("/upload-gnote")
+public class UploadGNoteServlet extends HttpServlet {
+
+  private final String COOKIE_NAME = "SFCookie";
+  private LabelService labelService = new LabelService();
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    if(!validateUser(request)) {
+      response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+      return;
+    }
+
+    HttpSession activeSession = request.getSession(false);
+    String userId = (String) activeSession.getAttribute("user_id");
+
+    String title = request.getParameter("title");
+    String school = request.getParameter("school").toLowerCase();
+    String course = request.getParameter("course").toLowerCase();
+    String sourceUrl = request.getParameter("sourceUrl");
+    String pdfSource = request.getParameter("pdfSource");
+
```

Comment:
Should we have a field in the chrome extension to add misc labels separated by commas? We already have the ability to handle this on the backend.

Reply:
The reason I didn't include it was because I didn't want to overwhelm the user with inputs in the chrome extension (since it's supposed to be an quick way to create a note), but if you think it'd be better to include it, I can add it.

Reply:
Ah I see, no that makes a lot of sense and I think you made the right decision. Was just wondering cause we could technically handle it but this is more of a UX thing.